### ISSUE TITLE
Infinite processing with 1.1.233 (never ends)

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#3261]
This is similar to when I reported https://github.com/microsoft/pyright/issues/2298

Just like then, it is affecting the same project (others are fine) and it seems to be weasyprint again. Here is a minimal code that reproduces the problem:

```python
import weasyprint

html = weasyprint.HTML(string="foobar")
html.write_pdf(target=None)
```

Running pyright on that code with `useLibraryCodeForTypes = true` will loop infinitely

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.234, which I just published. It will also be included in the next pylance release.
